File: Coding Problems & My Solutions/Jump Game/Jump_Game.java
class Solution {
    public boolean canJump(int[] nums) {
        if (nums.length < 2){
            return true;
        }
        int i = nums.length - 2;
        int n = 0;
        while (i >= 0){
            n++;
            if (nums[i] >= n){
                n = 0;
            }
            i--;
        }
        return n == 0;
    }
}

File: Coding Problems & My Solutions/Jump Game/Jump_Game.swift
class Solution {
    func canJump(_ nums: [Int]) -> Bool {
        var lastGoodIndex = nums.count - 1
        var i = nums.count - 1
        while i >= 0 {
            if i + nums[i] >= lastGoodIndex {
                lastGoodIndex = i
            }
            i -= 1
        }
        return lastGoodIndex == 0
    }
}